pvlib/data/gold/sdm.py

```diff
@@ -0,0 +1,832 @@
+            if g_sh == 0.:
+                iv_curve["r_sh"] = numpy.inf
+            else:
+                iv_curve["r_sh"] = 1. / g_sh
+            del iv_curve["g_sh"]
+
+            iv_curve["r_s"] = float.fromhex(iv_curve["r_s"])
+            iv_curve["nNsVth"] = float.fromhex(iv_curve["nNsVth"])
+            iv_curve["i_0"] = float.fromhex(iv_curve["i_0"])
+            iv_curve["i_l"] = float.fromhex(iv_curve["i_l"])
+            iv_curve["v_gold"] = numpy.array(
+                [float.fromhex(v) for v in iv_curve["v_gold"]])
+            iv_curve["i_gold"] = numpy.array(
+                [float.fromhex(i) for i in iv_curve["i_gold"]])
+
+    return gold_dataset_converted
+
+
+def pretty_print_gold_dataset(gold_dataset):
+    """
+    Create human-readable gold dataset.
+
+    From a gold dataset in the lossless, machine-interoperable format, create
+    and return the dataset as a human-readale, comma-separated-value (csv)
+    string.
+
+    Parameters
+    ----------
+    gold_dataset : dict
+
+    Returns
+    -------
+    gold_dataset_csv : str
+
+    """
+
+    # Convert gold dataset from hex
+    gold_dataset_converted = convert_gold_dataset(gold_dataset)
+
+    num_pts = len(
+        gold_dataset_converted["devices"][0]["iv_curves"][0]["v_gold"])
+
+    # Initialize output
+    gold_dataset_csv = gold_dataset_converted["timestamp_utc_iso"] + \
+        ", " + gold_dataset_converted["dataset_version"] + "\n\n"
+    gold_dataset_csv = gold_dataset_csv + "G (W/m^2), T (degC)" + "\n"
+    gold_dataset_csv = gold_dataset_csv + \
+        "r_sh (Ohm), r_s (Ohm), nNsVth (V), i_0 (A), i_l (A)" + "\n"
+    gold_dataset_csv = gold_dataset_csv + \
+        ", ".join(["v_gold_" + str(idx) +
+                   " (V)" for idx in range(num_pts)]) + "\n"
+    gold_dataset_csv = gold_dataset_csv + \
+        ", ".join(["i_gold_" + str(idx) +
+                   " (A)" for idx in range(num_pts)]) + "\n"
+    gold_dataset_csv = gold_dataset_csv + "\n\n"
+
+    for device in gold_dataset_converted["devices"]:
+        for iv_curve in device["iv_curves"]:
+            # Load G, T, and 5 single-diode model parameters for each I-V curve
+            # Floats are converted from lossless, interchangeable hex format
+            gold_dataset_csv = gold_dataset_csv + ", ".join(
+                map(str, [iv_curve["G"], iv_curve["T"]])) + "\n"
+            gold_dataset_csv = gold_dataset_csv + ", ".join(
+                map(str, [
+                    iv_curve["r_sh"], iv_curve["r_s"], iv_curve["nNsVth"],
+                    iv_curve["i_0"], iv_curve["i_l"]
+                ])) + "\n"
+            gold_dataset_csv = gold_dataset_csv + ", ".join(
+                map(str, iv_curve["v_gold"].tolist())) + "\n"
+            gold_dataset_csv = gold_dataset_csv + ", ".join(
+                map(str, iv_curve["i_gold"].tolist())) + "\n"
+
+    return gold_dataset_csv
+
+
+def save_gold_dataset(gold_dataset,
+                      json_rel_path="sdm.json",
```

Comment:
``os.path.abspath`` will work with either absolute or relative paths, so I think ``json_path`` is preferable. And similar for csv.

Reply:
Updated.